core/include/assert.h

```diff
@@ -48,6 +48,14 @@ extern "C" {
 #define RIOT_FILE_RELATIVE  (__FILE__)
 #endif
 
+#ifndef __NORETURN
```

Comment:
Suggesting (not sure why GitHub suggestions don't work right now):
```
/* Duplicating the definitions of kernel_defines.h as these are unsuitable for system header files like the assert.h currently being provided: kernel_defines.h would define symbols that are not reserved. */
#ifndef __NORETURN
```

although I would prefer if this was moved to a `core/include/kernel_defines_system.h` for reuse by `kernel_defines.h`

Either way, squashing will make this reviewable better.